managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/cache/PendingReadsManager.java

```diff
@@ -37,8 +39,21 @@
 /**
  * PendingReadsManager tries to prevent sending duplicate reads to BK.
  */
+@Slf4j
 public class PendingReadsManager {
 
+    /**
+     * Overhead per-entry to take into account the envelope.
+     */
+    private static final long BOOKKEEPER_READ_OVERHEAD_PER_ENTRY = 64;
+
+    private static final Gauge ESTIMATED_ENTRY_SIZE = Gauge
+            .build()
+            .name("pulsar_ml_pending_reads_estimated_entry_size")
+            .labelNames("topic")
+            .help("Estimated entry size for pending reads for the topic")
+            .register();
```

Comment:
We need to give users a way to disable topic level metrics. I'm not sure how to do that with this static metric, but I think it is an important requirement for clusters with a large number of topics.

Reply:
Good point.
I will update the patch

Reply:
@michaeljmarshall I have removed this metric.
it is not easy to pass the configuration to the PendingReadsManager

also, here we have a managed-ledger, that is it not always strictly a "topic"
there is no much value in this metrics